test/speed_regression/array.cpp

```diff
@@ -70,7 +69,6 @@ void test(size_t mb, size_t repeats) {
 	std::cout << "tpie::array" << std::endl;
 	blah_constructed = copy_constructed = std_constructed = assigned = 0;
 	{
-		boost::progress_timer _;
```

Comment:
I think the point of this file is that `boost::progress_timer` measures the elapsed time and outputs it on cout in the destructor.

I think we should either implement our own `progress_timer` and use that here, or just kill this file altogether.